### PR TITLE
Codecov how-to in CI section

### PR DESCRIPTION
**Pull request by ewallace**
- Explain how to use Codecov in coverage / continuous integration section.
- Added brief "What is CI" section

As discussed with @maelle and @mpadge in
https://discuss.ropensci.org/t/staff-admin-team-codecov/2982/3
https://github.com/ropensci/tidyqpcr/pull/191

### CODE REVIEW

File: pkg_building.Rmd [hunk 1]
```diff
@@ -326,7 +326,7 @@ For more explanations around licensing, refer to the [R packages book](https://r
     
 * If your package interacts with a database you might find [dittodb](https://docs.ropensci.org/dittodb) useful.
 
-* Once you've set up [continuous interaction (CI)](#ci), use your package's code coverage report (cf [this section of our book](#coverage)) to identify untested lines, and to add further tests.
+* Once you've set up [continuous integration (CI)](#ci), use your package's code coverage report (cf [this section of our book](#coverage)) to identify untested lines, and to add further tests.
```
Comment (mpadge) [inline pkg_building.Rmd:329]: good catch - thank you!

File: pkg_ci.Rmd [hunk 2]
```diff
@@ -6,9 +6,20 @@ This chapter summarizes our guidelines about continuous integration after explai
 Along with [last chapter](#building), it forms our guidelines for Software Peer Review.
 ```
 
+## What is continuous integration (CI)?
+
+Continuous integration automatically runs tests on software.
+In the case of rOpenSci, CI practically means that a set of tests will be automatically run via github, every time that you make a commit or pull request to github.
```
Comment (mpadge) [inline pkg_ci.Rmd:12]:
```suggestion
In the case of rOpenSci, CI practically means that a set of tests will be automatically run via GitHub, every time that you make a commit or pull request to GitHub.
```

File: pkg_ci.Rmd [hunk 3]
```diff
@@ -6,9 +6,20 @@ This chapter summarizes our guidelines about continuous integration after explai
...
+
+For CI to be meaningful you have to have tests set up that work locally using `R CMD check`, see [testing](/building.html#testing).
```
Comment (mpadge) [inline pkg_ci.Rmd:14]: Can this maybe be rephrased to a more pragmatic word than "meaningful"? Suggestion only: Something along the lines of "CI helps automate general package checks such as `R CMD check`. One of the main uses is testing ..."

File: pkg_ci.Rmd [hunk 4]
```diff
@@ -6,9 +6,20 @@ This chapter summarizes our guidelines about continuous integration after explai
...
+It is possible to set up CI before your tests are written, then CI will run the tests as you commit them to the repository.
+
 ## Why use continuous integration (CI)?
 
-All rOpenSci packages must use one form of continuous integration. This ensures that all commits, pull requests and new branches are run through `R CMD check`. rOpenSci packages' continuous integration must also be linked to a code coverage service, indicating how many lines are covered by unit tests.
+All rOpenSci packages must use one form of continuous integration.
+This ensures that all commits, pull requests and new branches are run through `R CMD check`.
+Tests are required to pass before merging pull requests, which provides another layer of information about problems and protection against breaking your package.
```
Comment (mpadge) [inline pkg_ci.Rmd:21]: This is not technically a requirement, so I'd prefer that statement to be modified or removed. Temporary breakages can arise through external reasons, so we can not and do not have an absolute requirement of passing tests at all times. The requirement amounts to something more like, "The latest avialable measure of test coverage should be at least 75%." Of course we'd never phrase it that way, so just declare a minimal coverage requirement and presume it to be current, whatever that may actually mean in practice.

File: pkg_ci.Rmd [hunk 5]
```diff
@@ -75,8 +86,18 @@ We no longer transfer AppVeyor projects to ropensci AppVeyor account so after tr
...
+Continuous integration should also include reporting of test coverage via a testing service such as [Codecov](https://codecov.io/) or [Coveralls](https://coveralls.io/).
+
+We recommend using Codecov.
+To activate Codecov for your repo, run `usethis::use_github_action("test-coverage")` to create a file `.github/workflows/test-coverage.yaml`.
+You also need to give Codecov access to your github repository, see [Codecov quick start guide](https://docs.codecov.com/docs/quick-start) for how to set up access.
+Then add a Codecov status badge to the top of your README.md, see [Codecov status badges](https://docs.codecov.com/docs/status-badges).
+
+Currently, Codecov has access to all ropensci github repositories by default.
+When your repository is accepted and transferred to ropensci, Codecov access should transfer automatically.
+You will need to update the url of the badge to point to the rOpenSci-hosted repository.
```
Comment (mpadge) [inline pkg_ci.Rmd:98]:
```suggestion
You will need to update the URL of the badge to point to the rOpenSci-hosted repository.
```

File: pkg_ci.Rmd [hunk 6]
```diff
@@ -75,8 +86,18 @@ We no longer transfer AppVeyor projects to ropensci AppVeyor account so after tr
...
+
+For more details, see the [README for the **covr** package](https://github.com/r-lib/covr) for instructions, as well as `usethis::use_coverage()`.
```
Comment (mpadge) [inline pkg_ci.Rmd:100]:
```suggestion
For more details, see the [README for the **covr** package](https://github.com/r-lib/covr) for instructions, as well as [`usethis::use_coverage()`](https://usethis.r-lib.org/reference/use_coverage.html) and [`usethis::use_github_action()`](https://usethis.r-lib.org/reference/github_actions.html).
```

**Review by mpadge — CHANGES_REQUESTED** [review]
Thank you very much @ewallace for this clear improvement in both structure and wording. Our Dev Guide is intended to be a community effort, and we particularly appreciate this kind of direct engagement with its ongoing development. I've made a couple of minor suggestions, and only the one not-really-minor request for change. Happy to iterate and discuss further as you see fit.

**Comment by maelle** [comment]
Thank you! Two other comments: please add yourself to https://github.com/ropensci/dev_guide/blob/55eba7e9c6703c783f7ac1c3dd3ae27140e28c8f/index.Rmd#L72 and a news item to https://github.com/ropensci/dev_guide/blob/55eba7e9c6703c783f7ac1c3dd3ae27140e28c8f/appendix.Rmd#L6

File: pkg_ci.Rmd (hunk 3 above)
Comment (ewallace) [reply pkg_ci.Rmd]: Changed to:
> CI automates the running of general package checks such as `R CMD check`, see [testing](/building.html#testing).

File: appendix.Rmd [hunk 7]
```diff
@@ -2,6 +2,10 @@
 
 # NEWS {#booknews}
 
+## 0.8.1
+
+- 2022-06-23, Expanded explanation of Codecov, added @ewallace to contributor list (#484)
```
Comment (maelle) [inline appendix.Rmd:7]:
```suggestion
- 2022-06-23, Expanded explanation of Codecov, added `@ewallace` to contributor list (#484)
```

**Review by maelle — APPROVED** [review]
thanks so much!

**Pull request merged**